admin/client/src/state/breadcrumbs/BreadcrumbsActions.js

```diff
@@ -0,0 +1,40 @@
+import ACTION_TYPES from './BreadcrumbsActionTypes';
+
+/**
+ * Set selected changeset item
+ *
+ * @param {array} List of breadcrumbs to add, each of which is an object
+ * with a 'text' and optional 'href' property.
+ * @return {object}
+ */
+export function setBreadcrumbs(breadcrumbs) {
+  return {
+    type: ACTION_TYPES.SET_BREADCRUMS,
+    payload: { breadcrumbs },
+  };
+}
+
+/**
+ * Adds a breadcrumb to the end of the list
+ *
+ * @param {object} Breadcrumb to add. Has 'text' and optional 'href' property.
+ * @return {function}
+ */
+export function pushBreadcrumb(breadcrumb) {
```

Comment:
The problem is that gridfields attached to records won't know anything about the modeladmin that model is attached to. It won't always know the full context.

How about parent components have a "base" breadcrumbs that are pushed to child components (via properties, not states). Then I'll remove push / pop?